keps/sig-network/2136-np-version-status/README.md

```diff
@@ -0,0 +1,789 @@
+Kubernetes—i.e., [kubernetes/kubernetes], we require the following Release
+Signoff checklist to be completed.
+
+Check these off as they are completed for the Release Team to track. These
+checklist items _must_ be updated for the enhancement to be released.
+-->
+
+Items marked with (R) are required *prior to targeting to a milestone / release*.
+
+- [ ] (R) Enhancement issue in release milestone, which links to KEP dir in [kubernetes/enhancements] (not the initial KEP PR)
+- [ ] (R) KEP approvers have approved the KEP status as `implementable`
+- [ ] (R) Design details are appropriately documented
+- [ ] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input
+- [ ] (R) Graduation criteria is in place
+- [ ] (R) Production readiness review completed
+- [ ] Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+Because NetworkPolicy is implemented by components outside of the
+Kubernetes core, there is often a lag between when new NetworkPolicy
+features are defined and when they are widely implemented. This means
+that in some cases a user cannot know if a cluster will correctly
+process a given NetworkPolicy.
+
+Related to this, there is currently no way for a NetworkPolicy
+implementation (ie, the network plugin) to indicate the status of
+NetworkPolicy processing to the user who created the policy.
+
+This KEP attempts to solve both of these problems.
+
+(Although it is not technically required that NetworkPolicy be
+implemented by the network plugin, this is almost universally the case,
+and so I will refer to NetworkPolicy implementations as "network
+plugins" through the rest of this document.)
+
+## Motivation
+
+### Goals
+
+- Allow users to indicate what version of the NetworkPolicy featureset
+  they are making use of (eg, whether they require support for combined
+  `podSelector` and `namespaceSelector`).
+
+- Allow network plugins to indicate when they are aware that they cannot
+  correctly implement a given NetworkPolicy.
+
+- Allow network plugins to indicate when a NetworkPolicy has been fully
+  "programmed" into the cluster network, to allow clients (including
+  humans, scripts, and cluster components) to get more reliable behavior
+  when adding new policies.
+
+### Non-Goals
+
```

Comment:
...   probably a non goal around latency / network tech specific application of rules updates / as a status type? i.e. it might take 20 ms to apply the new iptables ruleset in calico... were probably not in the business of reporting that update info regularly per policy here.  when i saw "network policy versioning and status" i thought it might be along those lines.

Reply:
I'm not sure exactly what you mean here

Reply:
```
### Non-Goals
- Granular/Implementation specific status information such as latency for applying iptables rules for policies or Sync'ing APIServer information.
```

Reply:
oh one other, something like ?
```
### Non-goals
- <something about this not being a feature gate>
```
just brought up in the netpol meeting today by @andrewsykim